extensions/opentelemetry/deployment/src/test/java/io/quarkus/opentelemetry/deployment/interceptor/WithSpanInterceptorTest.java

```diff
@@ -179,6 +241,26 @@ public void spanChild() {
         public void spanRestClient() {
             spanRestClient.spanRestClient();
         }
+
+        @WithSpan(value = "withSpanAndUni")
+        public Uni<String> spanUni() {
+            return Uni.createFrom().item("hello Uni");
```

Comment:
I would duplicate all these to have truly async responses and not only items emitted immediately.

Something like:

```java
@Inject Vertx vertx;

public Uni<String> spanUniAsync() {
        return Uni.createFrom().emitter(e -> {
            vertx.setTimer(100, x -> e.complete("hello Uni");
        });
}
```

None of these run on a duplicated context, too, which might make propagation tricky. I recommend also testing with and without duplicated context.